Algorithm for printing the occurrence of the given number in the list/array using Binary Search

narayan2111 opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
I added a binary search technique to find out the occurrences of a particular element appear in given list of an array.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#414 
I added a binary search technique to find out the occurrences of a particular element appear in the given list of an array.
User needs to enter an array in an ascending order to find out the occurrence of an entered element by the user.
![Program Output](https://user-images.githubusercontent.com/66173390/90725050-27b45180-e2dd-11ea-827c-058cf6b6a162.jpg)

narayan2111 committed:
Added Binary Search Technique

Added Binary search technique to find out the number of occurrences a element appear in list of an array.

narayan2111 committed:
Added Binary Search Technique 

Added Binary Search Technique to find out the number of occurrences a particular element appear in an given array

iamrajiv commented:
@narayan2111 binary search is already implemented in C++ at https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Binary_Search.cpp

narayan2111 commented:
> @narayan2111 binary search is already implemented in C++ at https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Binary_Search.cpp

I know sir ,but this is not a binary search only ,the issue assigned to me was to find out the occurrences of a element of an array using binary search technique.

narayan2111 commented:
> > @narayan2111 binary search is already implemented in C++ at https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Binary_Search.cpp
> 
> I know sir ,but this is not a binary search only ,the issue assigned to me was to find out the occurrences of a element of an array using binary search technique.
I'm attaching a link of an issue assigned to me 

https://github.com/TesseractCoding/NeoAlgo/issues/414#issue-679656509

iamrajiv commented on C-Plus-Plus/ds/Binary_search_technique.cpp:
```diff
@@ -0,0 +1,71 @@
+	int first=0,last=n-1,result=-1;//first and last variables to store first and last element of array
+	while(first<=last)			//result variable to store the occurrence
+	{
+		int mid = (first+last)/2;
```
@narayan2111 please avoid this technique to find the mid element since this can cause an integer overflow.
use this `int mid = first + ((last - first) / 2);`

narayan2111 replied:
I updated the code. @iamrajiv

iamrajiv commented on C-Plus-Plus/search/Binary Search technique.cpp:
```diff
@@ -0,0 +1,77 @@
+	int first=0,last=n-1,result=-1;//first and last variables to store first and last element of array
+	while(first<=last)			//result variable to store the occurrence
+	{
+		int mid = (first+last)/2;
```
@narayan2111 update this too

narayan2111 replied:
done with this too. @iamrajiv

narayan2111 committed:
Update Binary_search_technique.cpp

updated the algorithms of finding the mid value so that integer overflow can be remove.

narayan2111 committed:
Update Binary Search technique.cpp

changed the algorithm to find the mid value to overcome the integer overflow

vatsalkesarwani12 commented on C-Plus-Plus/search/Binary Search technique.cpp:
```diff
@@ -0,0 +1,77 @@
+
+/**
+ * copyright 2020 @author narayan2111
```
remove this comment

vatsalkesarwani12 commented on C-Plus-Plus/search/Binary Search technique.cpp:
```diff
@@ -0,0 +1,77 @@
+
+/*
+c++ program to implement binary search technique
+
```
either keep this comment at the top or at the bottom of the source code

narayan2111 replied:
> either keep this comment at the top or at the bottom of the source code

Done with changing the order of comments.

vatsalkesarwani12 reviewed (changes requested):
why 2 cpp file

do the required changes

narayan2111 committed:
Update Binary Search technique.cpp

Added "Algorithm for printing the occurrence of the given number in the list/array using Binary Search" and changed the order of comments

narayan2111 committed:
Update Binary_search_technique.cpp

Changed in the order of Comments.

narayan2111 committed:
Update Binary Search technique.cpp

Removed author comments

narayan2111 commented:
> why 2 cpp file
> 
> do the required changes

done with the changes sir.
one file is in readme and another one is in the c-plus-plus/ds directory.

iamrajiv reviewed (changes requested):
@narayan2111 this program should be in `cp` folder not in `ds` please change

iamrajiv reviewed (changes requested):
@narayan2111 add time and space complexity for the program

narayan2111 committed:
Changed location

narayan2111 committed:
Merge branch 'master' of https://github.com/narayan2111/NeoAlgo into master

narayan2111 committed:
Changed Location

narayan2111 committed:
Changed Location

narayan2111 commented:
> @narayan2111 this program should be in `cp` folder not in `ds` please change

Done sir ,changed the location.

narayan2111 commented:
> @narayan2111 add time and space complexity for the program

@iamrajiv  added space and time complexity.

narayan2111 committed:
Added Sparse_matrix_Operation.py

narayan2111 commented:
@atarax665 @iamrajiv @plazzy99 @Abhijit2505   Do check my PR so that it can be merged early. I'm waiting for this to be added. 😃

narayan2111 committed:
Update README.md

Added Sparsh Matrix

narayan2111 committed:
Update README.md

narayan2111 committed:
Added Time and Space Complexity

narayan2111 committed:
Merge branch 'master' of https://github.com/narayan2111/NeoAlgo into master

narayan2111 committed:
Update README.md

vatsalkesarwani12 reviewed (changes requested):
one pull request for one issue

narayan2111 committed:
Sparse Matrix Removed

narayan2111 commented:
> one pull request for one issue

Done sir . @plazzy99 @iamrajiv @atarax665 , now you can review it.

vatsalkesarwani12 reviewed (changes requested):
Currently, I can see 2 CPP file and 1 deleted file and I cannot find a readme file

narayan2111 committed:
Updated README.md

narayan2111 commented:
> Currently, I can see 2 CPP file and 1 deleted file and I cannot find a readme file

Done with the Updated of README file and deleted extra cpp file @plazzy99 @iamrajiv @atarax665

narayan2111 committed:
changes in comment heading statement.

narayan2111 committed:
Removed Binary Search Algorithm from Search directory

narayan2111 committed:
Merge branch 'master' into master

iamrajiv committed:
formatted code

narayan2111 commented:
@plazzy99 @atarax665 sir do check this pull request, so that i can take another issue also.
